### PR TITLE
RavenDB-12634 Database change vector isn't updated correctly when upd…

### PR DESCRIPTION
**Pull request by karmeli87**
…ating a counter

### CODE REVIEW

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Server/Documents/CountersStorage.cs [hunk 1]
```diff
@@ -251,9 +251,7 @@ private void PutCounterImpl(DocumentsOperationContext context, string documentId
 
                     if (changeVector == null)
                     {
-                        changeVector = ChangeVectorUtils
-                            .TryUpdateChangeVector(_documentDatabase.ServerStore.NodeTag, _documentsStorage.Environment.Base64Id, etag, string.Empty)
-                            .ChangeVector;
+                        changeVector = $"{_documentDatabase.ServerStore.NodeTag}:{etag}-{_documentsStorage.Environment.Base64Id}";
```
Comment (ayende) [inline src/Raven.Server/Documents/CountersStorage.cs]: Why build it manually?

File: src/Raven.Server/Documents/CountersStorage.cs [hunk 2]
```diff
@@ -370,9 +368,9 @@ public string IncrementCounter(DocumentsOperationContext context, string documen
                 RemoveTombstoneIfExists(context, documentId, name);
 
                 var etag = _documentsStorage.GenerateNextEtag();
-                var result = ChangeVectorUtils.TryUpdateChangeVector(_documentDatabase.ServerStore.NodeTag, _documentsStorage.Environment.Base64Id, etag, string.Empty);
+                var changeVector = $"{_documentDatabase.ServerStore.NodeTag}:{etag}-{_documentsStorage.Environment.Base64Id}";
```
Comment (ayende) [inline src/Raven.Server/Documents/CountersStorage.cs]: Would be better to move this to `ChangeVectorUtils` in general. Any changes / optimizations would only happen in one place

File: src/Sparrow/Json/LazyCompressedStringValue.cs [hunk 3]
```diff
@@ -126,5 +127,52 @@ public string Substring(int startIndex, int length)
         {
             return ToString().Substring(startIndex, length);
         }
+
```
Comment (ayende) [inline src/Sparrow/Json/LazyCompressedStringValue.cs:130]: Missing equality to `LazyStringValue`

File: src/Sparrow/Json/LazyCompressedStringValue.cs [hunk 4]
```diff
@@ -126,5 +127,52 @@ public string Substring(int startIndex, int length)
...
+            AllocatedMemoryData otherAllocated = null;
+            AllocatedMemoryData allocated = null;
+
+            try
+            {
+                var otherTempBuffer = other.DecompressToTempBuffer(out otherAllocated);
+                var tempBuffer = DecompressToTempBuffer(out allocated);
+
+                return Memory.Compare(tempBuffer, otherTempBuffer, UncompressedSize) == 0;
```
Comment (ayende) [inline src/Sparrow/Json/LazyCompressedStringValue.cs:166]: Comment here to explain different optimization levels can cause this

File: src/Raven.Server/Documents/CountersStorage.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/CountersStorage.cs]: done

File: src/Sparrow/Json/LazyCompressedStringValue.cs (hunk 3 above)
Comment (karmeli87) [reply src/Sparrow/Json/LazyCompressedStringValue.cs:130]: Will address it in a different issue: https://issues.hibernatingrhinos.com/issue/RavenDB-12731

**Review by ppekrol — APPROVED** [review]

**Pull request merged**